link.py

```diff
@@ -0,0 +1,115 @@
+from collections import deque
+'''
+   Class for a link.
+   
+   Attributes:
```

Comment:
use colons after each attribute to make it look better. like this,
link_id: a unique identifier for each link